--- nfs4-acl-tools-0.3.4/GUI/nfs4-acl-editor/nfs4acleditor.cpp.orig	2018-08-27 19:27:50.955855793 +0200
+++ nfs4-acl-tools-0.3.4/GUI/nfs4-acl-editor/nfs4acleditor.cpp	2018-08-27 19:35:23.829183952 +0200
@@ -569,7 +569,8 @@
 		if (i > strlen(ace->who)) {
 			dprintf("syncWho(): fromlen %d  tolen %u\n", strlen(ace->who), i);
 		}
-		strncpy(ace->who, who, NFS4_MAX_PRINCIPALSIZE);
+		strncpy(ace->who, who, NFS4_MAX_PRINCIPALSIZE-1);
+		ace->who[NFS4_MAX_PRINCIPALSIZE-1] = '\0';
 	}
 }
 
